spec/unit/events_collection/lockfile_spec.rb

```diff
@@ -0,0 +1,85 @@
+require_relative '../../../lib/events_collection/lockfile'
+
+describe 'Lockfile' do
+  let(:basepath) { '/tmp' }
+  let(:path) { File.join('/tmp', 'events_collection_run.lock') }
+  let(:new_lockfile) { CommonEvents::Lockfile.new(basepath) }
+
+  before(:each) do
+    File.delete(path) if File.exist?(path)
+  end
+
+  it 'sets the correct lockfile path' do
+    filepaths = ['/path/path', '/path/', '/path', 'path', '']
+    filepaths.each do |filepath|
+      expect(CommonEvents::Lockfile.new(filepath).filepath).to eq(File.join(filepath, '/events_collection_run.lock'))
+      File.delete(filepath) if File.exist?(filepath)
+    end
+  end
+
+  it 'requires a lockfile path' do
+    expect { CommonEvents::Lockfile.new }.to raise_error(ArgumentError)
+  end
+
+  it 'creates the lockfile with the correct format' do
+    new_lockfile.write_lockfile
+    expect(File.exist?(path)).to be true
+    lockfile_contents = JSON.parse(File.read(path))
+    expect(lockfile_contents['pid']).to be_an(Integer)
+    expect(lockfile_contents['program_name']).to be_a(String)
+  end
+
+  it 'checks if the lockfile exists' do
+    lockfile = new_lockfile
+    expect(lockfile.lockfile_exists?).to be false
+    lockfile.write_lockfile
+    expect(lockfile.lockfile_exists?).to be true
+  end
+
+  it 'can delete the lockfile' do
+    lockfile = new_lockfile
+    lockfile.write_lockfile
+    expect(File.exist?(path)).to be true
+    lockfile.remove_lockfile
+    expect(File.exist?(path)).to be false
+  end
+
+  it 'returns lockfile info correctly' do
+    lockfile = new_lockfile
+    expect(lockfile.info).to eql({ pid: nil, program_name: '' })
+    lockfile.write_lockfile
+    expect(lockfile.info['pid']).to be_an(Integer)
+    expect(lockfile.info['program_name']).to be_a(String)
+    File.delete(path)
+    expect(lockfile.info).to eql({ pid: nil, program_name: '' })
```

Comment:
Can we move this context below the two tests below? It seems like this is a set of tests unrelated to the tests around it which are all related to one another?

Reply:
You mean move the whole it block somewhere else, or split up the tests inside of the it block?

Reply:
This comment didn't attach to the line I expected it to for some reason. I mean the context `'checks if the process id is already running'` on line 57. The whole thing can be copy pasted below the two `it` blocks the follow it. That will keep those two `it`s attached to the rest of the it's above that context.